### PR TITLE
Feature/boas 778 metadata edit UI

### PR DESCRIPTION
**Pull request by lutamKC**
## Describe your changes

- New ui for editing metadata
- Changes on document properties page
- MEGA Refactor: `lib` and `component` have been merged into `common` and the `pages` folder has been removed
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/web/src/server/applications/case/documentation-metadata/__tests__/__snapshots__/documentation-metadata.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            </div>
+            <div id=\\"name-info\\" class=\\"govuk-hint govuk-character-count__message\\">You can enter up to 255 characters</div>
+        </div>
+        <div class=\\"govuk-grid-row\\">
+            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\">
+                <button class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save and return</button>
+            </div>
+        </div>
+    </form>
+</main>"
```
Comment (lutamKC) [inline apps/web/src/server/applications/case/documentation-metadata/__tests__/__snapshots__/documentation-metadata.test.js.snap:36]: everything else is a refactor, new code starts here

File: apps/web/src/server/views/applications/case-documentation/documentation-properties.njk [hunk 2]
```diff
@@ -1,113 +1,93 @@
...
-				<nav>
-					{% set actionLinks %}
-						<a class="govuk-link display--block" href="/open">Open</a>
-
-						<a class="govuk-link display--block" href="/move">Move</a>
-
-						<a class="govuk-link display--block" href="{{ 'document'|url({caseId: caseId, folderId: folderId, documentGuid: documentationFile.guid, step: 'delete'}) }}">Delete</a>
-					{% endset %}
-					{{ govukDetails({
+                <h2 class="govuk-caption-xl">Document properties</h2>
```
Comment (francescobrenci-AR) [inline apps/web/src/server/views/applications/case-documentation/documentation-properties.njk:22]: do you need an H1 on the page or is it inherited from the template

File: apps/web/src/server/views/applications/case-documentation/documentation-properties.njk [hunk 3]
```diff
@@ -1,113 +1,93 @@
...
-						{label: 'Webfilter', value: '', link: '#'},
-						{label: 'Date received', value: (documentationFile.receivedDate|datestamp({format: 'dd/MM/yyyy'})), link: '#'},
-						{label: 'Document type', value: '', link: '#'},
-						{label: 'Redaction', value: ('Redacted' if documentationFile.redacted else 'Unredacted') , link: '#'},
-						{label: 'Status', value: documentationFile.status|statusName , link: '#'},
-						{label: 'Exam library reference', value: '', link: '#'}
+                </nav>
+                <div class="govuk-!-margin-top-7">
+                    {% set properties = [
+						{label: 'File name', value: documentationFile.documentName|documentName, link: 'name'},
```
Comment (francescobrenci-AR) [inline apps/web/src/server/views/applications/case-documentation/documentation-properties.njk:54]: i'm assuming `documentationFile.[prop]` will need updating after integration with the API?

Comment (lutamKC) [reply apps/web/src/server/views/applications/case-documentation/documentation-properties.njk:54]: yes, at the moment only a few values actually exist

File: apps/web/src/server/views/applications/case-documentation/documentation-properties.njk (hunk 2 above)
Comment (lutamKC) [reply apps/web/src/server/views/applications/case-documentation/documentation-properties.njk]: no, H1 was missing. replaced the couple h2/h3 with the standard h1+caption

**Review by francescobrenci-AR — APPROVED** [review]

**Pull request merged**